### DISCUSSION TITLE
Jøkuluke 2023

### DISCUSSION THREAD

**Discussion by onordbo** [fremtind/jokul#2860] [Annet]
Åpner en tråd for en potensiell Jøkuluke, så vi får samlet litt behov, ideer, etc.

**Comment by onordbo** [comment]
Tema: Hvordan kommuniserer vi design intensjon før en løsning er laget? Bakgrunnsmateriale: https://github.com/fremtind/jokul/discussions/2557#discussioncomment-1773452

**Reply by gbrownlee** [reply]
Foreslår å bl.a. lage en "best i klassen"-løsning på et responsivitetscase som viser hvordan man kan og bør endre på bruk av komponenter og layout ved forskjellige vindusstørrelser.

**Comment by wkillerud** [comment]
**Kode i Jøkul-drakt**
Vi viser en del kodeeksempler i portalen (scroll f. eks. igjennom [guiden til å sette opp et nytt prosjekt](https://jokul.fremtind.no/guider/utvikler-nytt-prosjekt)), men annet enn minimale tweaks er det ikke våre farger som er involvert. Kan vi få til et fargetema for kode vi kan bruke i portalen? 

Bonuspoeng: kan vi lage et [tema for Visual Studio Code](https://code.visualstudio.com/api/extension-guides/color-theme) med utgangspunkt i hva vi får til i portalen? 🚀

**Comment by wkillerud** [comment]
**Fiks datovelgeren**

Se #2436 og alle småbugsene vi ser i dagens DatePicker

**Reply by wkillerud** [reply]
Blir fika før Jøkuluka, #2887

**Comment by onordbo** [comment]
## Designsprederen
En utforskende sprint i kreativ bruk av Figma's Web API, med mål om å bygge en mekanisme for å enkelt dele og spre design-godbiter ut i organisasjonen via TV-skjermene vi har rundt om i etasjene. Her er [inspirasjon og underlaget](https://www.figma.com/blog/introducing-figmas-platform/) som manifesterte frem ideen. Det ligger også inspirasjon til andre eksempelprosjekter vi kan lage vår egen vri på 💫

**Comment by onordbo** [comment]
## Figma Plugin

Jeg vet ikke hva denne nøyaktig skal inneholde enda, men det hadde vært kult å utforske mulighetene for hvordan vi kan gjøre Figma-hverdagen og Jøkul-bruken enklere gjennom vår egen plugin. Hva hadde vært kult å utforske? Tips til bruk og innsetting av Jøkul komponenter? Enkel tilgang på realistisk-ish testdata/tekster til bruk i skisser? Egen design-linter som sjekker om en skisse holder vår designstandard? Automatisk innsetting av bilder fra vår egen bildebank? Mulighetene her er i hovedsak kun begrenset av egen fantasi.

Eksempel: Netflix har laget sin egen [hendige Figma plugin](https://youtu.be/nFe6US9aA_U?t=1807) for å kunne teste ekte/realstisk innhold rett fra innholdsbiblioteket sitt i designskissene _(både tekst og bilder, på flere språk)_.

<img width="1988" alt="image" src="https://user-images.githubusercontent.com/4263132/171384624-73adb694-2cda-44e0-9025-fa2fc356fc36.png">

**Reply by leon-ho** [reply]
Jeg bruker Content Reel, f.eks. når jeg trenger å sette inn realistiske navn og fødselsnummer.

**Comment by MonaHolm72** [comment]
Utforske animasjon i løsningene våre

**Reply by onordbo** [reply]
+1 for å utforske animasjon. Om det er noen som har begynt å tenke på dette allerede, kan dere gjerne fylle på med ideer, inspirasjonskilder, behov og underlag i tråden her: https://github.com/fremtind/jokul/discussions/2923.

**Comment by MonaHolm72** [comment]
Moduler og bruk av redaksjonelle virkemidler på Fremtind.no

**Comment by wkillerud** [comment]
Hold av uke 5 (30. januar til 3. februar 2023) folkens

**Comment by MonaHolm72** [comment]
Oppdatere designprinsipper og knytte merkevareplattform opp mot Jøkul

**Comment by MonaHolm72** [comment]
Prinsipper for felles header og footer for kundeløsningene våre

**Comment by efatland** [comment]
### Visjonær Prototype av Hvordan Vi Helst Vil Bruke Jøkul

Vi bygger en prototype av en fiktiv fremtidig Fremtind-løsning. Løsningen skal være bra, designet skal være bra, koden skal være bra. Og vi slipper å bry oss om hva NICE får til eller produktsjefen har krevd. Og til sammen skal det vise hvor vi ønsker å ta Fremtinds digitale design og hvordan vi ønsker å bruke Jøkul.

En repeat av prototypen vi bygde for å utforske tokenization vs innramming.

**Comment by leon-ho** [comment]
### Compact

Lage retningslinjer for bruk av **compact**.

Noen diskusjonspunkter:
* Bør alt være compact i hele løsningen? (I Flyt har vi satt et flagg for at alt blir compact by default)
* Hvis noe er compact, og annet ikke er compact, blir folk forvirra (over at man har zoomet inn?)
* Man bør ta et bevisst valg når man velger å bruke compact i en komponent?
* Compact fungerer bedre i informasjonstett kontekst, e.g. i en tabell.

**Reply by piofinn** [reply]
Foreslår også å undersøke kompakt _layout_ versus kompakte/små _komponenter_. Forutsetter det ene det andre? Ikke? Hvordan brukes de sammen eller hver for seg?

**Comment by SercanSercan** [comment]
Enklere DatePicker
Datepicker ble en av de mest problematikk komponenter i Jøkul. Grunnen til dette kan være overkomplisert UI. Vi kan gå over alle bug oppgaver som har løst noe feil i DatePicker i siste året og beskrive smerte punkter. Dette arbeid kan føre til en enklere implementasjon av DatePicker i Jøkuluka.

**Reply by piofinn** [reply]
Brannfakkel: Trenger man i det hele tatt en datepicker? Kan man bruke [et annet mønster](https://design-system.service.gov.uk/components/date-input/) for å innhente datoer? Kanskje en liten behovsanalyse er noe man bør starte med her?

**Reply by KarolinaVangsnes** [reply]
kontroversielt

**Comment by leon-ho** [comment]
### Combobox / Multiselect

Trenger dette etterhvert, kanskje vi kan ferdigstille denne komponenten i løpet av Jøkuluka? 🤔

**Reply by ivnai** [reply]
Dette må vi få på plass🥳

**Comment by Stakset** [comment]
Utforske hvordan morgendagens skjemaløsning ser ut. En løsning som oppleves naturlig, kul og varm. Vi kan feks ta utgangspunkt i SeOpp sin skjemaløsning og lage et konsept som ligger nærmere ønsket kundeopplevelse.

**Reply by KarolinaVangsnes** [reply]
Og kanskje inkludere en innholdsdesigner også! Fordi språk er også en del av designsystemet! 🔥🔥🔥

**Reply by Stakset** [reply]
Det hadde vært gull å jobbe fokusert med Hilde i en uke 🥳

**Reply by gbrownlee** [reply]
Jeg snakket med Hilde. Hun kommer til å sette av det hun kan av tid for å være med. Forslag til konkrete og kanskje kortere miniworkshops hvor hun involveres tas imot med takk.
